cmd/virt-handler/virt-handler.go

```diff
@@ -150,6 +153,28 @@ func main() {
 	ws := new(restful.WebService)
 	ws.Route(ws.GET("/api/v1/namespaces/{namespace}/vms/{name}/console").To(console.Console))
 	restful.DefaultContainer.Add(ws)
+
+	labelSelector, err := labels.Parse("daemon in (virt-handler)")
+	if err != nil {
+		fmt.Printf("Can't parse label selector: %s\n", err.Error())
+		return
+	}
+	pods, err := coreClient.CoreV1().Pods(k8sv1.NamespaceDefault).List(meta_v1.ListOptions{LabelSelector: labelSelector.String()})
```

Comment:
Two things here:

1) I think that this query does not differentiate between the pods from different nodes. Which pod you would annotate is not defined this way.
2) I think we can simplify this a little bit. We already pass in the node-name as a parameter to virt-handler. When you look at `manifests/virt-handler.yaml.in`, we pass it to the pod like this:

```yaml
        env:
          - name: NODE_NAME
            valueFrom:
              fieldRef:
                fieldPath: spec.nodeName
```

and then reference it like this:

```yaml
          - "--hostname-override"
          - "$(NODE_NAME)"
```

It is also possible to pass in the `namespace` and the pod `name` with the same pattern. See [this](https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/) doc for the exact details. Then, after the self detection can shrink to something like this:

```golang
err := coreClient.Pods(namespace).Patch(podName, ...)
```